### PR TITLE
Revamp colors page

### PR DESCRIPTION
**Pull request by janfaracik**
Colors page

This PR revamps the colors page to add the new semantic and palette colors introduced in https://github.com/jenkinsci/jenkins/pull/6538.

It also adds an option to copy the color variable and class name by hovering over the color.

---

If anybody has any tips for cleaning up the Java side/adding internationalisation I'm all ears.

Thanks.

### CODE REVIEW

**Review by NotMyFault — COMMENTED** [review]
The four tips look quite different to me:

Currently testing on chrome with your core PR.

**Comment by janfaracik** [comment]
> The four tips look quite different to me:
> 
> Currently testing on chrome with your core PR.

Should be fixed now! Hadn't realised Chrome required a prefix for mask-image :)

File: src/main/java/io/jenkins/plugins/designlibrary/Colors.java [hunk 1]
```diff
@@ -12,14 +13,124 @@ public String getIconFileName() {
...
 
-    public List<String> getColors() {
-        // Uncomment these once these colors are in core
-        // return Arrays.asList("red", "green", "orange", "yellow", "blue", "indigo", "violet");
-        return Arrays.asList("red", "green", "orange");
+    public List<Semantic> getSemantics() {
+        return Arrays.asList(
+                new Semantic("Success", "Use for success messages", "success-color"),
+                new Semantic("Warning", "Use for warning messages", "warning-color"),
+                new Semantic("Error", "Use for error messages", "error-color"),
```
Comment (uhafner) [inline src/main/java/io/jenkins/plugins/designlibrary/Colors.java:20]: *messages* seems a bit too restrictive for me, in my plugins I will show line or bar charts, or an icon with this semantic colors as well. (I don't know an English word that makes a broader range for the usage clear, maybe you have an idea? Or use something like *messages, states, items*)

Comment (janfaracik) [reply src/main/java/io/jenkins/plugins/designlibrary/Colors.java]: Agreed 100%.  'States' is the first thing that came to my mind in response to your comment, I've updated it the wording :) Thanks!

**Review by timja — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
Nice to see my exclamation marks are back.

I've no issues invoking `mvn clean package` locally, but the CI fails on this task: https://github.com/jenkinsci/design-library-plugin/pull/55/commits/f1c35576ba387c4c87827e9eabd2c8f96e67cea2#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R11

**Comment by timja** [comment]
> but the CI fails on this task

Windows CI fails 😢

**Comment by janfaracik** [comment]
> > but the CI fails on this task
> 
> Windows CI fails 😢

Ah looks like Windows doesn't support `package.json` variables the same way macOS/Linux do :/ https://stackoverflow.com/questions/43705195/how-can-i-use-variables-in-package-json - I've removed them.

**Comment by timja** [comment]
still fails 😢

**Comment by NotMyFault** [comment]
Same failure like before, `--replace` seems to be the issue for the CI.

Though, I've no issue with `npm run mvnbuild` on sass 1.51.0 compiled with dart2js 2.16.2 locally. 
Maybe there's a version mismatch on the CI runner?

**Comment by janfaracik** [comment]
🎉 Needed `&&` rather than `;` doh

**Comment by timja** [comment]
Probably should’ve waited till the core PR is released, no need to revert but this page will be broken without it

**Pull request merged**